packages/itwinui-css/src/dialog/dialog.scss

```diff
@@ -51,11 +51,48 @@
   }
 }
 
-@mixin iui-dialog-default {
-  .iui-dialog {
-    left: 50%;
+@mixin iui-dialog-position {
+  top: 33%;
+  left: 50%;
+  transform: translate(-50%, -33%);
+
+  &[data-position] {
```

Comment:
Why you decided to use `data-position` instead of regular class as it is done in Toast?

Reply:
@mayank99 suggested it to me and I like the idea.  I would like to see us use `data-position` for toasts eventually too.  Is there any negative reason why we shouldn't do this?

Reply:
i suggested `data-placement` btw. `position` doesn't make sense to me. we might also want to add our prefix, e.g. `data-iui-placement`.

but yeah, the benefit of data attribute is that you can kinda enforce that only one of the options is active at a time. also has a nice JS API

Reply:
I can't think of any negative reasons besides consistency.